aspnetcore/web-api/action-return-types.md

```diff
@@ -36,60 +36,105 @@ This document explains when it's most appropriate to use each return type.
 
 The simplest action returns a primitive or complex data type (for example, `string` or a custom object type). Consider the following action, which returns a collection of custom `Product` objects:
 
-[!code-csharp[](../web-api/action-return-types/samples/WebApiSample.Api.21/Controllers/ProductsController.cs?name=snippet_Get)]
+[!code-csharp[](../web-api/action-return-types/samples/2x/WebApiSample.Api.21/Controllers/ProductsController.cs?name=snippet_Get)]
 
 Without known conditions to safeguard against during action execution, returning a specific type could suffice. The preceding action accepts no parameters, so parameter constraints validation isn't needed.
 
-When known conditions need to be accounted for in an action, multiple return paths are introduced. In such a case, it's common to mix an [ActionResult](/dotnet/api/microsoft.aspnetcore.mvc.actionresult) return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action.
+When known conditions need to be accounted for in an action, multiple return paths are introduced. In such a case, it's common to mix an <xref:Microsoft.AspNetCore.Mvc.ActionResult> return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action.
+
+::: moniker range=">= aspnetcore-3.0"
+
+### Return IAsyncEnumerable\<T>
+
+In ASP.NET Core 3.0 or later, a web API action can return <xref:System.Collections.Generic.IAsyncEnumerable%601>. The `IAsyncEnumerable<T>` support addresses the common mistake of returning an `IEnumerable<T>` type from an action. Most of .NET Standard's [strongly typed collection types](/dotnet/api/system.collections.generic) implement `IEnumerable<T>`. A common example of such a collection type is <System.Collections.Generic.List%601>. An action that returns an `IEnumerable<T>` type results in blocking because the serializer synchronously iterates the collection. 
```

Comment:
This isn't really an accurate description. It's not a bug to return `IEnumerable<>`. 

The problem is when you return an `IEnumerable<>` that **requires async work**.